### ISSUE TITLE
The relationship between the measurement value of the VQC circuit and the predicted probability

### ISSUE THREAD

**Issue by beyond997** [Qiskit/qiskit-machine-learning#602]
As a newcomer to quantum machine learning, I have some confusion and I hope to get help from everyone.

In the VQC tutorial for binary classification problems, for the measurement of the variational circuit, I believe it only measures one qubit. If the measurement is performed on pauliZ, what is the result returned by the measurement? (Is it the probability of each of the two states or an expected value?)

I also noticed that VQC uses the cross-entropy function as the loss function. The input of cross-entropy should be a probability value. Is it using the probability of the quantum state or the probability converted from the expected value?

Thank you very much

**Comment by adekusar-drl** [comment]
Hello, if you mean this [tutorial](https://qiskit.org/ecosystem/machine-learning/tutorials/02_neural_network_classifier_and_regressor.html), then in the VQC section all qubits are measured. VQC relies on the Sampler primitive that returns quasi-probabilities of measured bitstrings and then VQC maps these bitstrings to classes. Since we deal with probabilities a cross entropy loss is applied. Hope that helps.

**Comment by beyond997** [comment]
> Hello, if you mean this [tutorial](https://qiskit.org/ecosystem/machine-learning/tutorials/02_neural_network_classifier_and_regressor.html), then in the VQC section all qubits are measured. VQC relies on the Sampler primitive that returns quasi-probabilities of measured bitstrings and then VQC maps these bitstrings to classes. Since we deal with probabilities a cross entropy loss is applied. Hope that helps.

Thank you very much for your answer, it has refreshed my understanding. I used to think that only the first qubit was measured, so thank you for reminding me.

I have another small question. You mentioned that the cross-entropy uses the probabilities returned by the Sampler primitive, so the parameter shift rule needs to use these probabilities to calculate the gradient. However, when I browsed the literature on the parameter shift rule, I found that the gradient is calculated using the expected value of the circuit. I am a bit confused, so I would like to know the relationship between the expected value and probability in the Sampler primitive. I guess the two can be converted. I hope to get your answer again, and I am very grateful. @adekusar-drl

**Comment by adekusar-drl** [comment]
By default, gradients are evaluated by [ParamShiftSamplerGradient](https://qiskit.org/documentation/stubs/qiskit.algorithms.gradients.ParamShiftSamplerGradient.html). All required conversions/evaluations are handled by this class.

**Comment by beyond997** [comment]
> By default, gradients are evaluated by [ParamShiftSamplerGradient](https://qiskit.org/documentation/stubs/qiskit.algorithms.gradients.ParamShiftSamplerGradient.html). All required conversions/evaluations are handled by this class.
Okay, thank you for your guidance. I will read it carefully

**Comment by beyond997** [comment]
> By default, gradients are evaluated by [ParamShiftSamplerGradient](https://qiskit.org/documentation/stubs/qiskit.algorithms.gradients.ParamShiftSamplerGradient.html). All required conversions/evaluations are handled by this class.

I am very sorry to bother you again @adekusar-drl. I carefully examined the code for ParamShiftSamplerGradient below and found that the gradient is indeed calculated by the difference in probabilities, without utilizing any expected  

# Compute the gradients.
        gradients = []
        partial_sum_n = 0
        for n in all_n:
            gradient = []
            result = results.quasi_dists[partial_sum_n : partial_sum_n + n]   #results is a SamplerResult object, which contains probabilities or quasi-probabilities of bitstrings, plus optional metadata like error bars in the samples.
            for dist_plus, dist_minus in zip(result[: n // 2], result[n // 2 :]):
                grad_dist = defaultdict(float)
                for key, val in dist_plus.items():
                    grad_dist[key] += val / 2
                for key, val in dist_minus.items():
                    grad_dist[key] -= val / 2
                gradient.append(dict(grad_dist))
            gradients.append(gradient)
            partial_sum_n += n

However, in literature[1] , the gradient is calculated by the difference in expected values. So, is calculating the gradient using probabilities consistent with calculating it using expected values? Is there any literature related to this? I hope my question is not too much trouble for you. Thank you a lot!

[1] [Evaluating analytic gradients on quantum hardware](https://doi.org/10.1103/PhysRevA.99.032331)

**Comment by adekusar-drl** [comment]
@a-matsuo can you please take a look at the question above, do you have any literature handy?
@beyond997 If a quantum ML model is built on top of `Estimator` that evaluates an observable, then gradients are evaluated for expectation values.

**Comment by a-matsuo** [comment]
Hi @beyond997
It seems that you might be confused about the difference between `ParameterShiftSamplerGradient` and `ParameterShiftEstimatorGradient`. While both calculate gradients, they do so in different ways. In `ParameterShiftEstimatorGradient`, the gradient is obtained by calculating the difference between the expectation values, which may be more familiar to you. On the other hand, in `ParameterShiftSamplerGradient`, the gradient is calculated based on the difference of the quasiprobabilities of each bitstring.

If you are interested in understanding the difference between the two methods, you could try running `ParameterShiftSamplerGradient` on your computer and comparing the results to those obtained using `ParameterShiftEstimatorGradient`. This may help you gain a better understanding of the two methods and their respective applications.

**Comment by beyond997** [comment]
Thank you for your answer. I will use them to make some attempts. @a-matsuo
